OpenRA.Mods.Common/Activities/Repair.cs

```diff
@@ -33,12 +33,15 @@ public Repair(Actor self, Actor host, WDist closeEnough)
 		{
 			this.host = Target.FromActor(host);
 			this.closeEnough = closeEnough;
-			repairsUnits = host.Info.TraitInfo<RepairsUnitsInfo>();
+			repairsUnits = host.Trait<RepairsUnits>();
 			health = self.TraitOrDefault<Health>();
 		}
 
 		public override Activity Tick(Actor self)
 		{
+			if (repairsUnits.IsTraitDisabled)
```

Comment:
I'm just making a note here that the next line is `return NextActivity` and that the other changes have been made.